api_tests/base/test_auth.py

```diff
@@ -195,3 +196,23 @@ def test_node_write_scope_cant_read_user_view(self, mock_user_info):
 
         res = self.app.get(url, params=payload, auth='some_valid_token', auth_type='jwt', expect_errors=True)
         assert_equal(res.status_code, 403)
+
+    @mock.patch('framework.auth.cas.CasClient.profile')
+    def test_full_read_scope_can_read_guid_view(self, mock_user_info):
+        project = ProjectFactory()
+        mock_user_info.return_value = self._scoped_response(['osf.full_read'])
+        url = api_v2_url('guids/{}/'.format(project._id), base_route='/', base_prefix='v2/')
+        res = self.app.get(url, auth='some_valid_token', auth_type='jwt')
+        redirect_url = '{}{}nodes/{}/'.format(API_DOMAIN, API_BASE, project._id)
+        assert_equal(res.status_code, 302)
+        assert_equal(res.location, redirect_url)
```

Comment:
Since this request relies on a header, you'll probably want to verify that the whole transaction works (i.e. you can access the thing you're trying to access via the redirect). It can be an additional res and set of assertions, but I can see where just getting the URL won't necessarily get you the proper access to the resource.